siatune/ray/config.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) SI-Analytics. All rights reserved.
+# Modified from https://github.com/ray-project/ray/blob/ray-2.1.0/python/ray/train/torch/config.py  # noqa
+# for applying MM based repo training
+
+import logging
+import os
+from dataclasses import dataclass
+
+import ray
+import torch.distributed as dist
+from ray.train._internal.utils import get_address_and_port
+from ray.train._internal.worker_group import WorkerGroup
+from ray.train.backend import BackendConfig
+from ray.train.constants import DEFAULT_NCCL_SOCKET_IFNAME
+from ray.train.torch.config import _TorchBackend
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class CustomBackendConfig(BackendConfig):
+    """Configuration for torch process group setup."""
+
+    @property
+    def backend_cls(self):
+        return _CustomTorchBackend
+
+
+def _set_nccl_network_interface():
+    """Set the appropriate NCCL network interface to use."""
+
+    if 'NCCL_SOCKET_IFNAME' not in os.environ:
+        logger.debug(
+            f'Setting NCCL_SOCKET_IFNAME to {DEFAULT_NCCL_SOCKET_IFNAME} '
+            'to prioritize ethernet connection. To override this behavior, '
+            'set the `NCCL_SOCKET_IFNAME` environment variable in your Ray '
+            'runtime environment: '
+            "`ray.init(runtime_env={{'env_vars': {'NCCL_SOCKET_IFNAME': 'ens5'}})`"  # noqa
+        )
+        os.environ['NCCL_SOCKET_IFNAME'] = DEFAULT_NCCL_SOCKET_IFNAME
+
+
+class _CustomTorchBackend(_TorchBackend):
+    share_cuda_visible_devices: bool = True
+
+    def on_start(self, worker_group: WorkerGroup,
```

Comment:
If the process group is not initialized, how about initializing it here without throwing an error?

Reply:
I think we have no way to know whether the process group is initialized in `Task.run` method.